cmd/rpcdaemon/cli/config.go

```diff
@@ -557,6 +568,97 @@ func StartRpcServer(ctx context.Context, cfg httpcfg.HttpCfg, rpcAPI []rpc.API)
 	return nil
 }
 
+func startAuthenticatedRpcServer(ctx context.Context, cfg httpcfg.HttpCfg, rpcAPI []rpc.API) error {
+	var engineListener *http.Server
+	var engineSrv *rpc.Server
+	var engineHttpEndpoint string
+
+	// register apis and create handler stack
+	httpEndpoint := fmt.Sprintf("%s:%d", cfg.HttpListenAddress, cfg.HttpPort)
+
+	fmt.Printf("TraceRequests = %t\n", cfg.TraceRequests)
+	srv := rpc.NewServer(cfg.RpcBatchConcurrency, cfg.TraceRequests, cfg.RpcStreamingDisable)
+
+	allowListForRPC, err := parseAllowListForRPC(cfg.RpcAllowListFilePath)
+	if err != nil {
+		return err
+	}
+	srv.SetAllowList(allowListForRPC)
+
+	var defaultAPIList []rpc.API
+	var engineAPI []rpc.API
+
+	for _, api := range rpcAPI {
```

Comment:
No need to parse those. Those are always “engine” and “eth”

Reply:
Does it mean we should have one rpc.API list and append only "eth" and "engine" to it?

Reply:
yep, and append regardless of other params

Reply:
okay, got it. And then what about flags?

Reply:
you only care for the new two flags here, the rest, like http.api and the allow-list, you just ignore for the auth endpoint.